src/theme/searcher/searcher.js

```diff
@@ -259,6 +259,17 @@ window.search = window.search || {};
         doc_urls = config.doc_urls;
         searchindex = elasticlunr.Index.load(config.index);
 
+        searchbar.removeAttribute("disabled");
+        searchbar.focus();
```

Comment:
I don't like this behavior.

When I click the search button on a (simulated) slow connection, the search box appears, but I can't type anything into it until after the search index is loaded.

We've gone to a lot of hassle to avoid that in Rustdoc, because it's annoying to use when it works that way <https://github.com/rust-lang/rust/pull/90333#issuecomment-953166825>. Instead:

- If the user types something into the search box but the index isn't loaded yet, show a loading indicator of some sort. Hide it after the search index loads.
- After the search index loads, check if the search box is visible and has text in it. If they all check out, run the search.

Reply:
Makes sense. I will add a spinner.